consensus/polybft/signer/public.go

```diff
@@ -49,82 +29,86 @@ func (p *PublicKey) MarshalJSON() ([]byte, error) {
 }
 
 // UnmarshalJSON implements the json.Marshaler interface.
-func (p *PublicKey) UnmarshalJSON(b []byte) error {
-	var d []byte
+func (p *PublicKey) UnmarshalJSON(jsonBytes []byte) error {
+	var bytes []byte
 
-	err := json.Unmarshal(b, &d)
+	err := json.Unmarshal(jsonBytes, &bytes)
 	if err != nil {
 		return err
 	}
 
-	pk, err := UnmarshalPublicKey(d)
+	pub, err := UnmarshalPublicKey(bytes)
 	if err != nil {
 		return err
 	}
 
-	p.p = pk.p
+	p.g2 = pub.g2
 
 	return nil
 }
 
-// UnmarshalPublicKey reads the public key from the given byte array
-func UnmarshalPublicKey(raw []byte) (*PublicKey, error) {
-	if len(raw) == 0 {
-		return nil, errors.New("cannot unmarshal public key from empty slice")
-	}
-
-	p, err := bn254.NewG2().FromBytes(raw)
-	if err != nil {
-		return nil, err
-	}
-
-	return &PublicKey{p: p}, nil
-}
-
 // ToBigInt converts public key to 4 big ints
-func (p PublicKey) ToBigInt() [4]*big.Int {
+func (p *PublicKey) ToBigInt() [4]*big.Int {
 	blsKey := p.Marshal()
-	res := [4]*big.Int{
+	if len(blsKey) != 128 {
```

Comment:
Can this ever happen? Note that you are not doing the same check in the other functions that convert an array to bytes. We should check he `Marshal` function down to the library and figure out the size of the output and if that size can change.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1161/commits/2eaa53c3fd8ce7ee85eecbe9582617aa0d1b1a89